examples/tty_conv.c

```diff
@@ -0,0 +1,161 @@
+/* PlanC (hubenchang0515@outlook.com) -- an example application
+ * that implements a custom conversation */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <termio.h>
+#include <security/pam_appl.h>
+
+/***************************************
+ * @brief echo off/on
+ * @param[in] fd file descriptor
+ * @param[in] off 1 - echo off，0 - echo on
+ * *************************************/
+static void echoOff(int fd, int off)
+{
+    struct termio tty;
+    (void) ioctl(fd, TCGETA, &tty);
+    if (off)
+    {
+        tty.c_lflag &= ~(ECHO | ECHOE | ECHOK | ECHONL);
+        (void) ioctl(fd, TCSETAF, &tty);
+    }
+    else
+    {
+        tty.c_lflag |= (ECHO | ECHOE | ECHOK | ECHONL);
+        (void) ioctl(fd, TCSETAW, &tty);
+    }
+}
+
+/***************************************
+ * @brief echo off stdin
+ * *************************************/
+static void echoOffStdin()
+{
+    echoOff(fileno(stdin), 1);
+}
+
+/***************************************
+ * @brief echo on stdin
+ * *************************************/
+static void echoOnStdin()
+{
+    echoOff(fileno(stdin), 0);
+}
+
+/***************************************
+ * @brief read a line input
+ * @return the input string
+ * *************************************/
+static char* readline()
+{
+    struct termio tty;
+    char input[PAM_MAX_RESP_SIZE];
+
+    flockfile(stdin);
+    int i = 0;
+    for (; i < PAM_MAX_RESP_SIZE; i++)
+    {
+        int ch = getchar_unlocked();
+        if (ch == '\n' || ch == '\r' ||ch == EOF)
+            break;
+        input[i] = ch;
+    }
+    funlockfile(stdin);
+    input[i] = '\0';
+
+    return (strdup(input));
+}
+
+/**************************************************
+ * @brief callback of PAM conversation
+ * @param[in] num_msg the count of message
+ * @param[in] msg PAM message
+ * @param[out] resp our response
+ * @param[in] appdata_ptr custom data passed by struct pam_conv.appdata_ptr
+ * @return state
+ * ************************************************/
+static int conversation(int num_msg, const struct pam_message** msg, struct pam_response **resp, void *appdata_ptr)
+{
+    // check the count of message
```

Comment:
Please avoid C++ style comments.

Reply:
what is C++ style comments? is it means i should write as this?
```c
/* check the count of message */
```

Reply:
yes